llvm/lib/CodeGen/BranchFolding.cpp

```diff
@@ -123,9 +124,6 @@ INITIALIZE_PASS(BranchFolderLegacy, DEBUG_TYPE, "Control Flow Optimizer", false,
 PreservedAnalyses BranchFolderPass::run(MachineFunction &MF,
                                         MachineFunctionAnalysisManager &MFAM) {
   MFPropsModifier _(*this, MF);
-  bool EnableTailMerge =
```

Comment:
Since you're removing it here, the NPM require the flag `requiresStructuredCFG()` to be incorporated here.

https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Passes/CodeGenPassBuilder.h#L1235